### PR TITLE
PoC: react router

### PR DESCRIPTION
**Pull request by oscard0m**

### CODE REVIEW

File: src/main.jsx [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
 import { BaseStyles, ThemeProvider } from "@primer/react";
 
-import { OctokitProvider } from "./components/octokit-provider.js";
-import createStore from "./lib/create-store.js";
-import App from "./App.jsx";
+import Root from "./routes/root.jsx";
 import "./index.css";
+import { RouterProvider, createBrowserRouter } from "react-router-dom";
+import Login from "./routes/login.jsx";
+import NewIssueForm from "./routes/new.jsx";
```
Comment (oscard0m) [inline src/main.jsx:11]: If I'm not mistaken we don't need to rely on `/pages/...` path for GitHub Pages anymore but something to test.

File: src/routes/login.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          You need to sign in in order
+          <br />
+          to submit an issue.
+        </Text>
+
+        <Button
+          size="large"
+          leadingIcon={MarkGithubIcon}
+          as="a"
+          href={`${backendBaseUrl}/api/github/oauth/login?redirectUrl=${`http://localhost:8888`}`}
```
Comment (oscard0m) [inline src/routes/login.jsx:36]: ⚠️ TODO: Before, we relied on the current `window.location`, but now it needs to point to the homepage (the route in charge of redirecting).

File: src/routes/root.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// @ts-check
+
+import { useContext } from "react";
+import { Box, Spinner } from "@primer/react";
+import { OctokitContext } from "../components/octokit-provider.js";
+import { useNavigate } from "react-router-dom";
+
+// TODO: rely on loaders and actions
+
```
Comment (oscard0m) [inline src/routes/root.jsx:9]: This route is only doing redirects and showing a loading spinner in the meantime but I expect this to be the homepage of Project Forms.

Comment (oscard0m) [reply src/routes/root.jsx:9]: In upcoming commits, it is recommended we move to `loaders` and `actions` instead, [as recommended by React Router docs](https://reactrouter.com/en/main/hooks/use-navigate):
image

File: src/routes/new.jsx [hunk 4]
```diff
@@ -139,6 +143,11 @@ export default function NewIssuePage() {
   const [submittedIssueUrl, setSubmittedIssueUrl] = useState("");
   const [projectData, setProjectData] = useState(null);
   const [accessError, setAccessError] = useState(null);
+  const navigate = useNavigate();
+
+  if (authState.type === "unauthenticated") {
+    navigate("/login");
+  }
```
Comment (oscard0m) [inline src/routes/new.jsx:150]: Instead of relying on state update for rendering the `<Login>`, we redirect to `/login`

File: src/routes/new.jsx [hunk 5]
```diff
@@ -389,7 +398,13 @@ export default function NewIssuePage() {
 
   return (
     <>
-      <Nav avatarUrl={authState.user?.avatar_url} onLogout={logout} />
+      <Nav
+        avatarUrl={authState.user?.avatar_url}
+        onLogout={() => {
+          logout();
+          navigate("/login");
```
Comment (oscard0m) [inline src/routes/new.jsx:405]: Instead of relying on state update for rendering the `<Login>`, we redirect to `/login`

File: src/main.jsx (hunk 1 above)
Comment (gr2m) [reply src/main.jsx:11]: this was unrelated to GitHub Pages, just a convention that @drwpow suggested.

File: src/main.jsx [hunk 6]
```diff
@@ -1,19 +1,48 @@
...
+      {
+        index: true,
+        element: <Root />,
+      },
+      {
+        path: "/login",
+        element: <Login />,
+      },
+      {
+        path: "/project-forms/:repo/projects/:project_number/issues/new",
```
Comment (gr2m) [inline src/main.jsx:30]:
```suggestion
        path: "/:owner/:repo/projects/:project_number/issues/new",
```

File: src/App.jsx [hunk 7]
```diff
@@ -1,85 +1,13 @@
...
+import createStore from "./lib/create-store.js";
+import { OctokitProvider } from "./components/octokit-provider.js";
 
+export default function App() {
   return (
-    <ErrorBoundary FallbackComponent={ErrorFallback}>
-      <NewIssuePage />
-    </ErrorBoundary>
+    <OctokitProvider store={createStore("octokit-provider")}>
+      <Outlet />
```
Comment (gr2m) [inline src/App.jsx:10]: from our discussion: if we wanted e.g. a global header UI element, it would go here

File: src/routes/login.jsx (hunk 2 above)
Comment (gr2m) [reply src/routes/login.jsx:36]: Maybe we can introduce a new variable `import.meta.env.VITE_FRONTEND_BASE_URL` to get the redirect URL

File: src/routes/new.jsx (hunk 4 above)
Comment (gr2m) [reply src/routes/new.jsx:150]: can we add an optional `?redirectTo` query parameter here?

**Review by gr2m — APPROVED** [review]
let's ship as is and then do follow ups

**Pull request merged**